cogs/meta.py

```diff
@@ -250,27 +253,61 @@ async def support(self, ctx):
 
 		await ctx.send(self.bot.support_link)
 
-	@commands.command(aliases=['source'])
-	async def code(self, ctx, *, command: str = None):
-		'''Show command code or get GitHub repo link.'''
+	@commands.command(name="source", aliases=["src", "code"])
+	async def source(self, ctx: commands.Context, source_item: str = None):
+		"""Get a github link to the source code of a command."""
 
-		if command is None:
-			await ctx.send('https://github.com/Run1e/AceBot')
+		if source_item is None:
+			await ctx.send(GITHUB_LINK)
 			return
 
-		command = self.bot.get_command(command)
+		# send a typing event while we get from github
+		# normally this would be an async with,
+		# but no reason to do so since it shouldn't take very long to do this code,
+		# therefore no reason to make it with.
+		# if its been long enough that it has to fire again we've errorer or send a response
+		await ctx.trigger_typing()
+
+		cmd: commands.Command = self.bot.get_command(source_item)
 
-		if command is None or command.hidden:
-			raise commands.CommandError('Couldn\'t find command.')
+		if cmd is None:
+			raise commands.CommandError("Couldn't find command.")
+		callback = cmd.callback
+		source_file = str(
+			Path(inspect.getsourcefile(callback)).relative_to(str(Path.cwd()))
+		)
+		if "internal" in source_file:
+			# hide all commands in cogs.ahk.internal
+			raise commands.CommandError("Couldn't find command.")
 
-		code = '\n'.join(line[1:] for line in inspect.getsource(command.callback).splitlines())
+		lines, first_line_no = inspect.getsourcelines(callback)
 
-		paginator = commands.Paginator(prefix='```py')
-		for line in code.replace('``', '`\u200b`').split('\n'):
-			paginator.add_line(line)
+		lines_extension = f"#L{first_line_no}-L{first_line_no+len(lines)-1}"
+		link = f"{GITHUB_LINK}/blob/{GITHUB_BRANCH}/{source_file}{lines_extension}"
+
+		# check for image permissions and if we don't have them then just send the link
+		if not ctx.channel.permissions_for(ctx.me) >= discord.Permissions(
```

Comment:
Make this the default case and drop the embed below. Info about a command is provided by the .help and .about commands, and is not needed here.

Reply:
Github recently made their embeds a bit large, so the embed was to give useful information and not be as huge as the new github link embeds, which don't give as much useful information.

![image](https://user-images.githubusercontent.com/71233171/115155337-7ae9ca00-a04d-11eb-9525-5a922834deba.png)

Reply:
Links can be wrapped in <link> to avoid them being embedded

Reply:
But when they're wrapped they just look a bit odd.

![image](https://user-images.githubusercontent.com/71233171/115155488-2a26a100-a04e-11eb-8bee-a1ecdcca4e80.png)